lib/cartodb/api/request.rb

```diff
@@ -0,0 +1,126 @@
+module CartoDB
+  module Api
+    class Request
+
+      extend Forwardable
+
+      attr_accessor :configuration
+
+      def_delegators :configuration, :protocol, :domain, :api_key, :account, :base_url, :version, :timeout
+
+      def initialize(configuration)
+        self.configuration = configuration
+        reset
+      end
+
+      def method_missing(method, *args)
+        @path_steps << method.downcase
+        @path_steps << args if args.length > 0
+        @path_steps.flatten!
+        self
+      end
+
+      def create(params = nil, headers = nil, body = nil)
+        make_request(:post, params, headers, body)
+      end
+
+      def update(params = nil, headers = nil, body = nil)
+        make_request(:patch, params, headers, body)
+      end
+
+      def retrieve(params = nil, headers = nil)
+        make_request(:get, params, headers)
+      end
+
+      def delete(params = nil, headers = nil)
+        make_request(:delete, params, headers)
+      end
+
+      protected
+
+      def reset
+        @path_steps = []
+      end
+
+      def client
+        faraday = Faraday.new(url: url) do |client|
+          client.params['api_key'] = api_key
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.